events/source.go

```diff
@@ -296,54 +348,81 @@ var _ BlockNotifier = &Notifier{}
 // Notifier can notify a consumer about new blocks
 type Notifier struct {
 	logger         log.Logger
+	getClient      BlockClientFactory
 	queryNotifier  *queryBlockNotifier
 	eventsNotifier *eventblockNotifier
 	running        context.Context
 	shutdown       context.CancelFunc
 	start          int64
+	timeout        time.Duration
 }
 
 // Done returns a channel that is closed when the Notifier has completed cleanup
-func (b Notifier) Done() <-chan struct{} {
+func (b *Notifier) Done() <-chan struct{} {
 	return b.eventsNotifier.Done()
 }
 
 // NewBlockNotifier returns a new BlockNotifier instance
+// NewBlockNotifier maintains compatibility with vald
 func NewBlockNotifier(client BlockClient, logger log.Logger, options ...DialOption) *Notifier {
+	getClient := func() (BlockClient, error) {
+		return client, nil
+	}
+	return NewBlockNotifierWithDialOptions(getClient, logger, options...)
+}
+
+// NewBlockNotifierWithDialOptions returns a new BlockNotifier instance with dial options
+func NewBlockNotifierWithDialOptions(getClient BlockClientFactory, logger log.Logger, options ...DialOption) *Notifier {
```

Comment:
this is a misnomer, the difference are bot the dialoptions but the client factory